### DISCUSSION TITLE
Developer setup: running single test file

### DISCUSSION THREAD

**Discussion by NPCRUS** [com-lihaoyi/scalasql#55] [Q&A] [answered]
Hi folks, I was trying to setup repo for contributing, immediately bumped into issue when trying to run specific test file, what I've tried:
- `mill __.testOnly scalasql.query.SchemaTests`
- `mill __.test scalasql.h2.query`
- `mill __.testOnly scalasql.h2.query.SchemaTests`
and other variations
the closest thing that works: `mill __.test scalasql.h2` but it's not enough for me
is there a way to do it? (I am using mill and utest for sometime already and I tried other documented ways of run single test file)

**Comment by lihaoyi** [comment, marked as answer]
I think it's `./mill __.test scalasql.h2.SchemaTests`; the second thing is the fully qualified name of the concrete `object` extending `TestSuite`, which is `scalasql.h2.SchemaTests`

**Comment by NPCRUS** [comment]
thanks, I almost wrote that i tried all possibilities, my intuition was right 🥲

**Comment by lihaoyi** [comment]
No worries, you probably got confused by the abstract `trait` or `class` versions of these suites, which look like the suites to run, but are actually just helpers to be instantiated in `ConcreteTestSuites.scala` as concrete objects which are the real test suites

**Reply by NPCRUS** [reply]
thanks for answering my next question
